File: calculateMode.js
exports.calculateMode = function(array) {
    let obj = {}
    let mode = 0
    let result = []
    for (let element of array) {
        if (!obj[element]) {
            obj[element] = 1
        } else {
            obj[element] += 1
        }
    }
    for (let char in obj) {
        if (obj[char] > mode) {
            mode = obj[char]
        }
    }
    for (let char in obj) {
        if (obj[char] == mode) {
            result.push(char)
        }
    }
    return result
}

File: calculate_mode.py
def calculate_mode(list):
    most = 0
    mode = []
    my_dict = {}
    for l in list:
        if l in my_dict:
            my_dict[l] += 1
        else: 
            my_dict[l] = 1
    for key, value in my_dict.items():
        if value > most:
            most = value
    for key, value in my_dict.items():
        if value == most:
            mode.append(key)
    return mode 
